---
layout: splash
permalink: /
header:
  overlay_color: "#5e616c"
  overlay_image: /assets/images/mm-home-page-feature.jpg
excerpt: "Computational Biologist. Bioinformatician at the Quadram Institute Biosciences."
intro: 
  - excerpt: "Below you will find links to other pages with code, documentation, and technical notes for bioinformatics and computational evolutionary biology software."

featr_code:
  - image_path: /assets/images/logo_tatajuba.png
    alt: "tatajuba"
    title: "Tatajuba"
    excerpt: "Identification and classification of homopolymeric tracts from reads"
    url: "https://github.com/quadram-institute-bioscience/tatajuba"
    btn_class: "btn--primary btn--small"
    btn_label: "source code on github"

  - image_path: /assets/images/logo_uvaia.png
    alt: "uvaia"
    title: "Uvaia"
    excerpt: "Reference-based alignment and search against aligned database" 
    url: "https://github.com/quadram-institute-bioscience/uvaia"
    btn_class: "btn--primary btn--small"
    btn_label: "source code on github"

  - image_path: /assets/images/logo_superdistance.png
    alt: "super_distance"
    title: "Super Distance"
    excerpt: "Supertree estimation from gene family trees, using matrix representation with distances"
    url: "https://github.com/quadram-institute-bioscience/super_distance"
    btn_class: "btn--primary btn--small"
    btn_label: "source code on github"

  - image_path: /assets/images/logo_peroba.png
    alt: "peroba"
    title: "Peroba"
    excerpt: "Phylogenetic exploration of SARS-CoV-2 sequences"
    url: "https://github.com/quadram-institute-bioscience/peroba"
    btn_class: "btn--primary btn--small"
    btn_label: "source code on github"

  - image_path: /assets/images/logo_guenomu.png
    alt: "guenomu"
    title: "Guenomu"
    excerpt: "Hierarchical Bayesian species tree estimation from multi-gene family data"
    url: "https://bitbucket.org/leomrtns/guenomu/"
    btn_class: "btn--primary btn--small"
    btn_label: "source code on bitbucket"

  - image_path: /assets/images/logo_genefam.png
    alt: "genefam_dist"
    title: "Genefam-dist"
    excerpt: "Signature inference of gene family trees from anchoring species trees."
    url: "https://github.com/leomrtns/genefam-dist"
    btn_class: "btn--primary btn--small"
    btn_label: "source code on github"

  - image_path: /assets/images/logo_specimage.png
    alt: "specimage"
    title: "SpecImage"
    excerpt: "Python module to work with hyperspectral imaging data."
    url: "https://github.com/leomrtns/specimage"
    btn_class: "btn--primary btn--small"
    btn_label: "source code on github"

  - image_path: /assets/images/logo_biomcmc.png
    alt: "biomcmclib"
    title: "biomcmc-lib library"
    excerpt: "C library with low level functions for phylogenetic analyses"
    url: "https://github.com/quadram-institute-bioscience/biomcmc-lib"
    btn_class: "btn--primary btn--small"
    btn_label: "source code on github"

  - image_path: /assets/images/logo_biomc2.png
    alt: "recomb-biomc2"
    title: "Recomb-biomc2"
    excerpt: "Hiearchical Bayesian Model for Viral Phylogenetic Recombination"
    url: "https://github.com/leomrtns/recomb-biomc2"
    btn_class: "btn--primary btn--small"
    btn_label: "source code on github"

  - image_path: /assets/images/logo_cumaru.png
    alt: "cumaru"
    title: "Cumaru"
    excerpt: "(Experimental) multiple sequence alignment based on kalign3"
    url: "https://github.com/leomrtns/cumaru"
    btn_class: "btn--primary btn--small"
    btn_label: "source code on github"

  - image_path: /assets/images/logo_curupixa.png
    alt: "curupixa"
    title: "Curupixa"
    excerpt: "(Experimental) thread-safe API-aware C library to supercede biomcmc-lib someday"
    url: "https://github.com/leomrtns/curupixa"
    btn_class: "btn--primary btn--small"
    btn_label: "source code on github"

featr_texts:
  - image_path: /assets/images/logo_araiguma.png
    alt: "notebooks"
    title: "Binfie notes"
    excerpt: "Blog with technical notes on bioinformatics algorithms and implementations"
    url: "https://leomrtns.github.io/jupyterblog"
    btn_class: "btn--info btn--small"
    btn_label: "Notebooks webpage"

  - image_path: /assets/images/logo_biomcmc.png
    alt: "biomcmc-libdocs"
    title: "biomcmc-lib API"
    excerpt: "Doxygen documentation for employing biomcmc-lib source code"
    url: "https://leomrtns.github.io/doxygen-biomcmclib"
    btn_class: "btn--info btn--small"
    btn_label: "API documentation page"

  - image_path: /assets/images/logo_biomcmc_blog.png
    alt: "biomcmc-blog"
    title: "bioMCMC blog"
    excerpt: "Old blog with phylogenetics observations and new papers"
    url: "http://biomcmc.blogspot.com"
    btn_class: "btn--info btn--small"
    btn_label: "read blog posts"

featr_suppl:
  - image_path: /assets/images/logo_salmobiofilm.png
    alt: "Salmonella biofilm"
    title: "Salmonella Biofilm"
    excerpt: "Jupyter notebooks with analyses of experimental evolution of Salmonella biofilms"
    url: "https://github.com/quadram-institute-bioscience/2020.Salmonella_biofilm"
    btn_class: "btn--warning btn--small"
    btn_label: "github repository"

  - image_path: /assets/images/logo_70s.png
    alt: "long-read rRNA phylogenetics"
    title: "rRNA phylogenetics"
    excerpt: "Analysis of the phylogenetic resolution of the rRNA operon"
    url: "https://github.com/quadram-institute-bioscience/70S-resolution"
    btn_class: "btn--warning btn--small"
    btn_label: "github repository"

  - image_path: /assets/images/logo_ucatest.png
    alt: "UCA test bias"
    title: "Tree homology tests"
    excerpt: "Critical evaluation of tree-based homology inference"
    url: "https://bitbucket.org/leomrtns/uca-test-bias"
    btn_class: "btn--warning btn--small"
    btn_label: "bitbucket repository"

separ_code:
  - image_path: /assets/images/h_separator.png
    excerpt: '<b class="notice--primary">Software and code</b>'
separ_texts:
  - image_path: /assets/images/h_separator.png
    excerpt: '<b class="notice--info">Documentation and Texts</b>'
separ_suppl:
  - image_path: /assets/images/h_separator.png
    excerpt: '<b class="notice--warning">Supplementary analyses for manuscripts</b>'
---

{% include feature_row id="intro" type="center" %}
{% include feature_row id="separ_code" type="center" %}
{% include feature_row id="featr_code" %}
{% include feature_row id="separ_suppl" type="center" %}
{% include feature_row id="featr_suppl" %} 
{% include feature_row id="separ_texts" type="center" %}
{% include feature_row id="featr_texts" %} 
<a rel="me" href="https://mstdn.science/@leomrtns">Mastodon</a>
