<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.13"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>biomcmc-lib: alignment.h File Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectlogo"><img alt="Logo" src="_biomcmclib-thumbnail.png"/></td>
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">biomcmc-lib
   &#160;<span id="projectnumber">0.1</span>
   </div>
   <div id="projectbrief">low level library for phylogenetic analysis</div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.13 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Search');
</script>
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
$(function() {
  initMenu('',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
</script>
<div id="main-nav"></div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="dir_97aefd0d527b934f1d99a682da8fe6a9.html">lib</a></li>  </ul>
</div>
</div><!-- top -->
<div class="header">
  <div class="summary">
<a href="#nested-classes">Data Structures</a> &#124;
<a href="#typedef-members">Typedefs</a> &#124;
<a href="#func-members">Functions</a>  </div>
  <div class="headertitle">
<div class="title">alignment.h File Reference</div>  </div>
</div><!--header-->
<div class="contents">

<p>File handling functions and calculation of distances for sequence data in nexus format.  
<a href="#details">More...</a></p>
<div class="textblock"><code>#include &quot;<a class="el" href="distance__matrix_8h_source.html">distance_matrix.h</a>&quot;</code><br />
<code>#include &quot;<a class="el" href="nexus__common_8h_source.html">nexus_common.h</a>&quot;</code><br />
</div><div class="textblock"><div class="dynheader">
Include dependency graph for alignment.h:</div>
<div class="dyncontent">
<div class="center"><img src="alignment_8h__incl.png" border="0" usemap="#alignment_8h" alt=""/></div>
<map name="alignment_8h" id="alignment_8h">
<area shape="rect" id="node2" href="distance__matrix_8h.html" title="distance matrix, that can be used in alignments and trees, and patristic&#45;distance based species dista..." alt="" coords="417,155,542,181"/>
<area shape="rect" id="node23" href="nexus__common_8h.html" title="File handling functions for nexus format in general. " alt="" coords="578,80,703,107"/>
<area shape="rect" id="node3" href="hashtable_8h.html" title="double hashing open&#45;address hash table using strings as key â€“ also has distance matrix, that can be used in alignments and trees " alt="" coords="549,528,639,555"/>
<area shape="rect" id="node4" href="hashfunctions_8h.html" title="Collections of hash functions for 32 and 64 bits, including one&#45;liners, murmurhash, and xxhash. " alt="" coords="547,603,662,629"/>
<area shape="rect" id="node5" href="bipartition_8h.html" title="Unary/binary operators on arbitrarily&#45;sized bitstrings (strings of zeros and ones) like split biparti..." alt="" coords="576,677,665,704"/>
<area shape="rect" id="node6" href="lowlevel_8h.html" title="Lowest level header file. Header file for lowlevel.c. " alt="" coords="669,752,749,779"/>
<area shape="rect" id="node24" href="char__vector_8h.html" title="list of strings (each string is a vector of chars) " alt="" coords="654,229,755,256"/>
<area shape="rect" id="node28" href="empirical__frequency_8h.html" title="Creates a histogram of a vector, ordered by frequency. " alt="" coords="566,155,715,181"/>
<area shape="rect" id="node25" href="prob__distribution_8h.html" title="Probability distribution functions and auxiliary mathematical functions from statistical package R..." alt="" coords="594,304,722,331"/>
<area shape="rect" id="node26" href="random__number_8h.html" title="Random number generation, with algorithms from the Gnu Scientific Library (GSL) and motivation from t..." alt="" coords="537,379,664,405"/>
<area shape="rect" id="node27" href="random__number__gen_8h_source.html" title="random_number_gen.h" alt="" coords="517,453,671,480"/>
</map>
</div>
</div><div class="textblock"><div class="dynheader">
This graph shows which files directly or indirectly include this file:</div>
<div class="dyncontent">
<div class="center"><img src="alignment_8h__dep__incl.png" border="0" usemap="#alignment_8hdep" alt=""/></div>
<map name="alignment_8hdep" id="alignment_8hdep">
<area shape="rect" id="node2" href="kmerhash_8h.html" title="k&#45;mer handling of DNA sequences, with hash transformation " alt="" coords="5,80,96,107"/>
<area shape="rect" id="node3" href="biomcmc_8h.html" title="biomcmc library interface to external programs, specific to super_sptree repo. " alt="" coords="7,155,94,181"/>
</map>
</div>
</div>
<p><a href="alignment_8h_source.html">Go to the source code of this file.</a></p>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="nested-classes"></a>
Data Structures</h2></td></tr>
<tr class="memitem:"><td class="memItemLeft" align="right" valign="top">struct &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="structalignment__struct.html">alignment_struct</a></td></tr>
<tr class="memdesc:"><td class="mdescLeft">&#160;</td><td class="mdescRight">Data from alignment file.  <a href="structalignment__struct.html#details">More...</a><br /></td></tr>
<tr class="separator:"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="typedef-members"></a>
Typedefs</h2></td></tr>
<tr class="memitem:ab92d2b81028457d293c6e56b2c2b815c"><td class="memItemLeft" align="right" valign="top"><a id="ab92d2b81028457d293c6e56b2c2b815c"></a>
typedef struct <a class="el" href="structalignment__struct.html">alignment_struct</a> *&#160;</td><td class="memItemRight" valign="bottom"><b>alignment</b></td></tr>
<tr class="separator:ab92d2b81028457d293c6e56b2c2b815c"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="func-members"></a>
Functions</h2></td></tr>
<tr class="memitem:aa036d9585db0e7f896efff4e37ffd5cb"><td class="memItemLeft" align="right" valign="top"><a id="aa036d9585db0e7f896efff4e37ffd5cb"></a>
<a class="el" href="structalignment__struct.html">alignment</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="alignment_8h.html#aa036d9585db0e7f896efff4e37ffd5cb">read_alignment_from_file</a> (char *seqfilename)</td></tr>
<tr class="memdesc:aa036d9585db0e7f896efff4e37ffd5cb"><td class="mdescLeft">&#160;</td><td class="mdescRight">Reads DNA alignment (guess format between FASTA and NEXUS) from file and store info in <a class="el" href="structalignment__struct.html" title="Data from alignment file. ">alignment_struct</a>. <br /></td></tr>
<tr class="separator:aa036d9585db0e7f896efff4e37ffd5cb"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a57980b340fee11061ad4511efc7b7c47"><td class="memItemLeft" align="right" valign="top"><a id="a57980b340fee11061ad4511efc7b7c47"></a>
<a class="el" href="structalignment__struct.html">alignment</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="alignment_8h.html#a57980b340fee11061ad4511efc7b7c47">read_fasta_alignment_from_file</a> (char *seqfilename)</td></tr>
<tr class="memdesc:a57980b340fee11061ad4511efc7b7c47"><td class="mdescLeft">&#160;</td><td class="mdescRight">Reads DNA FASTA alignment from file and store info in <a class="el" href="structalignment__struct.html" title="Data from alignment file. ">alignment_struct</a>. <br /></td></tr>
<tr class="separator:a57980b340fee11061ad4511efc7b7c47"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a5bd5335c2fd2fcf8c30e1becfea64b2c"><td class="memItemLeft" align="right" valign="top"><a id="a5bd5335c2fd2fcf8c30e1becfea64b2c"></a>
<a class="el" href="structalignment__struct.html">alignment</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="alignment_8h.html#a5bd5335c2fd2fcf8c30e1becfea64b2c">read_nexus_alignment_from_file</a> (char *seqfilename)</td></tr>
<tr class="memdesc:a5bd5335c2fd2fcf8c30e1becfea64b2c"><td class="mdescLeft">&#160;</td><td class="mdescRight">Reads DNA NEXUS alignment from file and store info in <a class="el" href="structalignment__struct.html" title="Data from alignment file. ">alignment_struct</a>. <br /></td></tr>
<tr class="separator:a5bd5335c2fd2fcf8c30e1becfea64b2c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae7be6382b75167950cff01c19dbbe166"><td class="memItemLeft" align="right" valign="top"><a id="ae7be6382b75167950cff01c19dbbe166"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="alignment_8h.html#ae7be6382b75167950cff01c19dbbe166">print_alignment_in_fasta_format</a> (<a class="el" href="structalignment__struct.html">alignment</a> align, FILE *stream)</td></tr>
<tr class="memdesc:ae7be6382b75167950cff01c19dbbe166"><td class="mdescLeft">&#160;</td><td class="mdescRight">Prints alignment to FILE stream in FASTA format (debug purposes). <br /></td></tr>
<tr class="separator:ae7be6382b75167950cff01c19dbbe166"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a98e39e61ac259230b72c60d7f033bdd4"><td class="memItemLeft" align="right" valign="top"><a id="a98e39e61ac259230b72c60d7f033bdd4"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="alignment_8h.html#a98e39e61ac259230b72c60d7f033bdd4">del_alignment</a> (<a class="el" href="structalignment__struct.html">alignment</a> align)</td></tr>
<tr class="memdesc:a98e39e61ac259230b72c60d7f033bdd4"><td class="mdescLeft">&#160;</td><td class="mdescRight">Frees memory from <a class="el" href="structalignment__struct.html" title="Data from alignment file. ">alignment_struct</a>. <br /></td></tr>
<tr class="separator:a98e39e61ac259230b72c60d7f033bdd4"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a9f784a5ee1238ede5538c085b40cfbca"><td class="memItemLeft" align="right" valign="top"><a id="a9f784a5ee1238ede5538c085b40cfbca"></a>
<a class="el" href="structdistance__matrix__struct.html">distance_matrix</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="alignment_8h.html#a9f784a5ee1238ede5538c085b40cfbca">new_distance_matrix_from_valid_matrix_elems</a> (<a class="el" href="structdistance__matrix__struct.html">distance_matrix</a> original, int *valid, int n_valid)</td></tr>
<tr class="memdesc:a9f784a5ee1238ede5538c085b40cfbca"><td class="mdescLeft">&#160;</td><td class="mdescRight">new matrix of pairwise distance by simply excluding original elements not present in valid[] <br /></td></tr>
<tr class="separator:a9f784a5ee1238ede5538c085b40cfbca"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a51ff9ec227fbe812471f316a55bea52b"><td class="memItemLeft" align="right" valign="top"><a id="a51ff9ec227fbe812471f316a55bea52b"></a>
<a class="el" href="structdistance__matrix__struct.html">distance_matrix</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="alignment_8h.html#a51ff9ec227fbe812471f316a55bea52b">new_distance_matrix_from_alignment</a> (<a class="el" href="structalignment__struct.html">alignment</a> align)</td></tr>
<tr class="memdesc:a51ff9ec227fbe812471f316a55bea52b"><td class="mdescLeft">&#160;</td><td class="mdescRight">creates and calculates matrix of pairwise distances based on alignment <br /></td></tr>
<tr class="separator:a51ff9ec227fbe812471f316a55bea52b"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a0d5562e7469a65589c80e20dc561e3ce"><td class="memItemLeft" align="right" valign="top"><a id="a0d5562e7469a65589c80e20dc561e3ce"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="alignment_8h.html#a0d5562e7469a65589c80e20dc561e3ce">store_likelihood_info_at_leaf</a> (double **l, char *align, int n_pat, int n_state)</td></tr>
<tr class="memdesc:a0d5562e7469a65589c80e20dc561e3ce"><td class="mdescLeft">&#160;</td><td class="mdescRight">transform aligned sequence into likelihood for terminal taxa (e.g. A -&gt; 0001, C-&gt; 0010 etc) (e.g. A -&gt; 0001, C-&gt; 0010 etc) (e.g. A -&gt; 0001, C-&gt; 0010 etc) (e.g. A -&gt; 0001, C-&gt; 0010 etc) (e.g. A -&gt; 0001, C-&gt; 0010 etc) (e.g. A -&gt; 0001, C-&gt; 0010 etc) (e.g. A -&gt; 0001, C-&gt; 0010 etc) (e.g. A -&gt; 0001, C-&gt; 0010 etc) <br /></td></tr>
<tr class="separator:a0d5562e7469a65589c80e20dc561e3ce"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><p>File handling functions and calculation of distances for sequence data in nexus format. </p>
<p>Reading of sequence data in nexus format (sequencial or interleaved) and fasta format. For fasta format the sequences don't need to be aligned, but for all formats if the sequences are aligned a data compression is used so that we keep only the distinct site (column) patterns and a mapping between original and compressed site columns. Based on the sequence pairs we can also calculate the matrix of distances between sequences. </p>
</div></div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.13
</small></address>
</body>
</html>
