<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.13"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>biomcmc-lib: distance_matrix.h File Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectlogo"><img alt="Logo" src="_biomcmclib-thumbnail.png"/></td>
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">biomcmc-lib
   &#160;<span id="projectnumber">0.1</span>
   </div>
   <div id="projectbrief">low level library for phylogenetic analysis</div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.13 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Search');
</script>
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
$(function() {
  initMenu('',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
</script>
<div id="main-nav"></div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="dir_97aefd0d527b934f1d99a682da8fe6a9.html">lib</a></li>  </ul>
</div>
</div><!-- top -->
<div class="header">
  <div class="summary">
<a href="#nested-classes">Data Structures</a> &#124;
<a href="#typedef-members">Typedefs</a> &#124;
<a href="#func-members">Functions</a>  </div>
  <div class="headertitle">
<div class="title">distance_matrix.h File Reference</div>  </div>
</div><!--header-->
<div class="contents">

<p>distance matrix, that can be used in alignments and trees, and patristic-distance based species distances  
<a href="#details">More...</a></p>
<div class="textblock"><code>#include &quot;<a class="el" href="hashtable_8h_source.html">hashtable.h</a>&quot;</code><br />
</div><div class="textblock"><div class="dynheader">
Include dependency graph for distance_matrix.h:</div>
<div class="dyncontent">
<div class="center"><img src="distance__matrix_8h__incl.png" border="0" usemap="#distance__matrix_8h" alt=""/></div>
<map name="distance__matrix_8h" id="distance__matrix_8h">
<area shape="rect" id="node2" href="hashtable_8h.html" title="double hashing open&#45;address hash table using strings as key – also has distance matrix, that can be used in alignments and trees " alt="" coords="664,80,753,107"/>
<area shape="rect" id="node3" href="hashfunctions_8h.html" title="Collections of hash functions for 32 and 64 bits, including one&#45;liners, murmurhash, and xxhash. " alt="" coords="651,155,766,181"/>
<area shape="rect" id="node4" href="bipartition_8h.html" title="Unary/binary operators on arbitrarily&#45;sized bitstrings (strings of zeros and ones) like split biparti..." alt="" coords="664,229,753,256"/>
<area shape="rect" id="node5" href="lowlevel_8h.html" title="Lowest level header file. Header file for lowlevel.c. " alt="" coords="669,304,749,331"/>
</map>
</div>
</div><div class="textblock"><div class="dynheader">
This graph shows which files directly or indirectly include this file:</div>
<div class="dyncontent">
<div class="center"><img src="distance__matrix_8h__dep__incl.png" border="0" usemap="#distance__matrix_8hdep" alt=""/></div>
<map name="distance__matrix_8hdep" id="distance__matrix_8hdep">
<area shape="rect" id="node2" href="alignment_8h.html" title="File handling functions and calculation of distances for sequence data in nexus format. " alt="" coords="511,80,601,107"/>
<area shape="rect" id="node5" href="topology__common_8h.html" title="General&#45;purpose topology structures created from nexus_tree_struct (and low&#45;level functions) ..." alt="" coords="181,80,320,107"/>
<area shape="rect" id="node16" href="distance__generator_8h.html" title="distance calculation between generic objects,without generating full matrix beforehand ..." alt="" coords="345,80,487,107"/>
<area shape="rect" id="node3" href="kmerhash_8h.html" title="k&#45;mer handling of DNA sequences, with hash transformation " alt="" coords="488,304,579,331"/>
<area shape="rect" id="node4" href="biomcmc_8h.html" title="biomcmc library interface to external programs, specific to super_sptree repo. " alt="" coords="317,453,403,480"/>
<area shape="rect" id="node6" href="topology__distance_8h.html" title="branch length operations on topologies, including patristic distances " alt="" coords="181,155,320,181"/>
<area shape="rect" id="node7" href="parsimony_8h.html" title="binary and multistate parsimony matrices, together with bipartition extraction for MRP ..." alt="" coords="313,379,407,405"/>
<area shape="rect" id="node8" href="topology__randomise_8h.html" title="Creation of random topologies and modification of existing ones through branch swapping. " alt="" coords="5,229,155,256"/>
<area shape="rect" id="node13" href="read__newick__trees_8h.html" title="Low&#45;level functions for reading newick strings. " alt="" coords="179,229,322,256"/>
<area shape="rect" id="node9" href="upgma_8h.html" title="UPGMA and bioNJ from (onedimensional representation of) distance matrices. " alt="" coords="49,304,121,331"/>
<area shape="rect" id="node10" href="genetree_8h.html" title="gene tree and species tree structures, for reconciliation etc. This is the high&#45;level file with globa..." alt="" coords="53,379,136,405"/>
<area shape="rect" id="node11" href="reconciliation_8h.html" title="low&#45;level file for gene tree and species tree reconciliation. This file is hidden from user and conta..." alt="" coords="40,453,149,480"/>
<area shape="rect" id="node12" href="splitset__distances_8h.html" title="Low&#45;level functions that use only the split bipartitions of topologies – treating them as unrooted u..." alt="" coords="25,528,164,555"/>
<area shape="rect" id="node14" href="newick__space_8h.html" title="Reads a list of trees in newick format and creates vector of topologies. " alt="" coords="181,304,297,331"/>
<area shape="rect" id="node15" href="topology__space_8h.html" title="Reads tree files in nexus format and creates a vector of topologies. " alt="" coords="165,379,289,405"/>
<area shape="rect" id="node17" href="clustering__goptics_8h.html" title="OPTICS algorithm based on https://github.com/guineri/GOPTICS. " alt="" coords="359,155,497,181"/>
</map>
</div>
</div>
<p><a href="distance__matrix_8h_source.html">Go to the source code of this file.</a></p>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="nested-classes"></a>
Data Structures</h2></td></tr>
<tr class="memitem:"><td class="memItemLeft" align="right" valign="top">struct &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="structdistance__matrix__struct.html">distance_matrix_struct</a></td></tr>
<tr class="separator:"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:"><td class="memItemLeft" align="right" valign="top">struct &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="structspdist__matrix__struct.html">spdist_matrix_struct</a></td></tr>
<tr class="separator:"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="typedef-members"></a>
Typedefs</h2></td></tr>
<tr class="memitem:a9e145719317f4d9e5067479ec7d53bb8"><td class="memItemLeft" align="right" valign="top"><a id="a9e145719317f4d9e5067479ec7d53bb8"></a>
typedef struct <a class="el" href="structdistance__matrix__struct.html">distance_matrix_struct</a> *&#160;</td><td class="memItemRight" valign="bottom"><b>distance_matrix</b></td></tr>
<tr class="separator:a9e145719317f4d9e5067479ec7d53bb8"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad771eef7a2693d94a03fb0b80df60664"><td class="memItemLeft" align="right" valign="top"><a id="ad771eef7a2693d94a03fb0b80df60664"></a>
typedef struct <a class="el" href="structspdist__matrix__struct.html">spdist_matrix_struct</a> *&#160;</td><td class="memItemRight" valign="bottom"><b>spdist_matrix</b></td></tr>
<tr class="separator:ad771eef7a2693d94a03fb0b80df60664"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="func-members"></a>
Functions</h2></td></tr>
<tr class="memitem:a05c728b75adbababd8c6fd4de241912a"><td class="memItemLeft" align="right" valign="top"><a id="a05c728b75adbababd8c6fd4de241912a"></a>
<a class="el" href="structdistance__matrix__struct.html">distance_matrix</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="distance__matrix_8h.html#a05c728b75adbababd8c6fd4de241912a">new_distance_matrix</a> (int nseqs)</td></tr>
<tr class="memdesc:a05c728b75adbababd8c6fd4de241912a"><td class="mdescLeft">&#160;</td><td class="mdescRight">creates new matrix of pairwise distances <br /></td></tr>
<tr class="separator:a05c728b75adbababd8c6fd4de241912a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a32656dd2ef988382d7e7a3eb505d1522"><td class="memItemLeft" align="right" valign="top"><a id="a32656dd2ef988382d7e7a3eb505d1522"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="distance__matrix_8h.html#a32656dd2ef988382d7e7a3eb505d1522">zero_lower_distance_matrix</a> (<a class="el" href="structdistance__matrix__struct.html">distance_matrix</a> dist)</td></tr>
<tr class="memdesc:a32656dd2ef988382d7e7a3eb505d1522"><td class="mdescLeft">&#160;</td><td class="mdescRight">specially in gene/sptree distance methods (GLASS, STEAC, etc.) lower is used for means and upper for min. This function resets matrix elements <br /></td></tr>
<tr class="separator:a32656dd2ef988382d7e7a3eb505d1522"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a5ac533926151f08c78923a645ab11446"><td class="memItemLeft" align="right" valign="top"><a id="a5ac533926151f08c78923a645ab11446"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="distance__matrix_8h.html#a5ac533926151f08c78923a645ab11446">transpose_distance_matrix</a> (<a class="el" href="structdistance__matrix__struct.html">distance_matrix</a> dist)</td></tr>
<tr class="memdesc:a5ac533926151f08c78923a645ab11446"><td class="mdescLeft">&#160;</td><td class="mdescRight">invert lower and upper diagonals of matrix (since some functions like upgma expect upper, etc.) <br /></td></tr>
<tr class="separator:a5ac533926151f08c78923a645ab11446"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6c66792124e982cdb5202e1be10c24cf"><td class="memItemLeft" align="right" valign="top"><a id="a6c66792124e982cdb5202e1be10c24cf"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="distance__matrix_8h.html#a6c66792124e982cdb5202e1be10c24cf">del_distance_matrix</a> (<a class="el" href="structdistance__matrix__struct.html">distance_matrix</a> dist)</td></tr>
<tr class="memdesc:a6c66792124e982cdb5202e1be10c24cf"><td class="mdescLeft">&#160;</td><td class="mdescRight">releases memory allocated to distance_matrix (this structure has no smart ref_counter) <br /></td></tr>
<tr class="separator:a6c66792124e982cdb5202e1be10c24cf"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a9837e1b9bded4fd9e565e714713c390c"><td class="memItemLeft" align="right" valign="top"><a id="a9837e1b9bded4fd9e565e714713c390c"></a>
<a class="el" href="structspdist__matrix__struct.html">spdist_matrix</a>&#160;</td><td class="memItemRight" valign="bottom"><b>new_spdist_matrix</b> (int n_species)</td></tr>
<tr class="separator:a9837e1b9bded4fd9e565e714713c390c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae266042e13a382a1890a6fbff681b9f2"><td class="memItemLeft" align="right" valign="top">void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="distance__matrix_8h.html#ae266042e13a382a1890a6fbff681b9f2">zero_all_spdist_matrix</a> (<a class="el" href="structspdist__matrix__struct.html">spdist_matrix</a> dist)</td></tr>
<tr class="separator:ae266042e13a382a1890a6fbff681b9f2"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a2be42a346c2a1ba84b8e5694e7eb9893"><td class="memItemLeft" align="right" valign="top"><a id="a2be42a346c2a1ba84b8e5694e7eb9893"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><b>finalise_spdist_matrix</b> (<a class="el" href="structspdist__matrix__struct.html">spdist_matrix</a> dist)</td></tr>
<tr class="separator:a2be42a346c2a1ba84b8e5694e7eb9893"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6ddbc5097dbe689d2c74ea5fbb56e173"><td class="memItemLeft" align="right" valign="top"><a id="a6ddbc5097dbe689d2c74ea5fbb56e173"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><b>finalise_spdist_matrix_with_rescaling</b> (<a class="el" href="structspdist__matrix__struct.html">spdist_matrix</a> dist, double scale)</td></tr>
<tr class="separator:a6ddbc5097dbe689d2c74ea5fbb56e173"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a3b0ce1bc6fc8e338ca04435fe22b4459"><td class="memItemLeft" align="right" valign="top"><a id="a3b0ce1bc6fc8e338ca04435fe22b4459"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><b>complete_missing_spdist_from_global_spdist</b> (<a class="el" href="structspdist__matrix__struct.html">spdist_matrix</a> local, <a class="el" href="structspdist__matrix__struct.html">spdist_matrix</a> global)</td></tr>
<tr class="separator:a3b0ce1bc6fc8e338ca04435fe22b4459"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad8672a57540a96d4437a05a775282abe"><td class="memItemLeft" align="right" valign="top"><a id="ad8672a57540a96d4437a05a775282abe"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><b>copy_spdist_matrix_to_distance_matrix_upper</b> (<a class="el" href="structspdist__matrix__struct.html">spdist_matrix</a> spd, <a class="el" href="structdistance__matrix__struct.html">distance_matrix</a> dist, <a class="el" href="lowlevel_8h.html#a97a80ca1602ebf2303258971a2c938e2">bool</a> use_means)</td></tr>
<tr class="separator:ad8672a57540a96d4437a05a775282abe"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a15093d39f1c6485ce09e1a7643ce6736"><td class="memItemLeft" align="right" valign="top"><a id="a15093d39f1c6485ce09e1a7643ce6736"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><b>del_spdist_matrix</b> (<a class="el" href="structspdist__matrix__struct.html">spdist_matrix</a> dist)</td></tr>
<tr class="separator:a15093d39f1c6485ce09e1a7643ce6736"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ad04d393e906484301674610590ae2dfe"><td class="memItemLeft" align="right" valign="top"><a id="ad04d393e906484301674610590ae2dfe"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="distance__matrix_8h.html#ad04d393e906484301674610590ae2dfe">fill_species_dists_from_gene_dists</a> (<a class="el" href="structdistance__matrix__struct.html">distance_matrix</a> spdist, <a class="el" href="structdistance__matrix__struct.html">distance_matrix</a> gendist, int *sp_id, <a class="el" href="lowlevel_8h.html#a97a80ca1602ebf2303258971a2c938e2">bool</a> use_upper_gene)</td></tr>
<tr class="memdesc:ad04d393e906484301674610590ae2dfe"><td class="mdescLeft">&#160;</td><td class="mdescRight">updates distances between species based on genes and gene-to-species mapping, with min on upper and mean on lower diagonal <br /></td></tr>
<tr class="separator:ad04d393e906484301674610590ae2dfe"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ab5c7f9dfd23d5fab609c82e86a9a7f3d"><td class="memItemLeft" align="right" valign="top"><a id="ab5c7f9dfd23d5fab609c82e86a9a7f3d"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="distance__matrix_8h.html#ab5c7f9dfd23d5fab609c82e86a9a7f3d">update_species_dists_from_spdist</a> (<a class="el" href="structdistance__matrix__struct.html">distance_matrix</a> global, <a class="el" href="structdistance__matrix__struct.html">distance_matrix</a> local, int *spexist)</td></tr>
<tr class="memdesc:ab5c7f9dfd23d5fab609c82e86a9a7f3d"><td class="mdescLeft">&#160;</td><td class="mdescRight">update global (over loci) species distances besed on local (within locus) species distances <br /></td></tr>
<tr class="separator:ab5c7f9dfd23d5fab609c82e86a9a7f3d"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a3302aeae38b037be76a360a5fb8d7898"><td class="memItemLeft" align="right" valign="top"><a id="a3302aeae38b037be76a360a5fb8d7898"></a>
int&#160;</td><td class="memItemRight" valign="bottom"><b>prepare_spdistmatrix_from_gene_species_map</b> (<a class="el" href="structspdist__matrix__struct.html">spdist_matrix</a> spdist, int *sp_id, int n_sp_id)</td></tr>
<tr class="separator:a3302aeae38b037be76a360a5fb8d7898"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a760f76e6dd06a82994c45dda3d450f37"><td class="memItemLeft" align="right" valign="top"><a id="a760f76e6dd06a82994c45dda3d450f37"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><b>fill_spdistmatrix_from_gene_dists</b> (<a class="el" href="structspdist__matrix__struct.html">spdist_matrix</a> spdist, <a class="el" href="structdistance__matrix__struct.html">distance_matrix</a> gendist, int *sp_id, <a class="el" href="lowlevel_8h.html#a97a80ca1602ebf2303258971a2c938e2">bool</a> use_upper_gene)</td></tr>
<tr class="separator:a760f76e6dd06a82994c45dda3d450f37"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac0bd82593abed577a8be45076f9697c6"><td class="memItemLeft" align="right" valign="top"><a id="ac0bd82593abed577a8be45076f9697c6"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="distance__matrix_8h.html#ac0bd82593abed577a8be45076f9697c6">fill_spdistmatrix_from_gene_dist_vector</a> (<a class="el" href="structspdist__matrix__struct.html">spdist_matrix</a> spdist, double *gdist, int n_gdist, int *sp_id)</td></tr>
<tr class="memdesc:ac0bd82593abed577a8be45076f9697c6"><td class="mdescLeft">&#160;</td><td class="mdescRight">initialise spdist_matrix with patristic distances from gdist vector of size n_gdist (1D) <br /></td></tr>
<tr class="separator:ac0bd82593abed577a8be45076f9697c6"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4610b2cfcc845942ae75c6f00e4d6bc5"><td class="memItemLeft" align="right" valign="top"><a id="a4610b2cfcc845942ae75c6f00e4d6bc5"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><b>update_spdistmatrix_from_spdistmatrix</b> (<a class="el" href="structspdist__matrix__struct.html">spdist_matrix</a> global, <a class="el" href="structspdist__matrix__struct.html">spdist_matrix</a> local)</td></tr>
<tr class="separator:a4610b2cfcc845942ae75c6f00e4d6bc5"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><p>distance matrix, that can be used in alignments and trees, and patristic-distance based species distances </p>
<p>These functions don't know about trees/topologies: topology_common.c creates the actual patristic distances, and downstream software or <a class="el" href="genetree_8h.html" title="gene tree and species tree structures, for reconciliation etc. This is the high-level file with globa...">genetree.h</a> should decide how to use this information </p>
</div><h2 class="groupheader">Function Documentation</h2>
<a id="ae266042e13a382a1890a6fbff681b9f2"></a>
<h2 class="memtitle"><span class="permalink"><a href="#ae266042e13a382a1890a6fbff681b9f2">&#9670;&nbsp;</a></span>zero_all_spdist_matrix()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void zero_all_spdist_matrix </td>
          <td>(</td>
          <td class="paramtype"><a class="el" href="structspdist__matrix__struct.html">spdist_matrix</a>&#160;</td>
          <td class="paramname"><em>dist</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">
<p>zero both mean[] and min[] since we only look at average (never min) across loci </p>

</div>
</div>
</div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.13
</small></address>
</body>
</html>
