<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.13"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>biomcmc-lib: genetree.h File Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectlogo"><img alt="Logo" src="_biomcmclib-thumbnail.png"/></td>
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">biomcmc-lib
   &#160;<span id="projectnumber">0.1</span>
   </div>
   <div id="projectbrief">low level library for phylogenetic analysis</div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.13 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Search');
</script>
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
$(function() {
  initMenu('',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
</script>
<div id="main-nav"></div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="dir_97aefd0d527b934f1d99a682da8fe6a9.html">lib</a></li>  </ul>
</div>
</div><!-- top -->
<div class="header">
  <div class="summary">
<a href="#nested-classes">Data Structures</a> &#124;
<a href="#typedef-members">Typedefs</a> &#124;
<a href="#func-members">Functions</a>  </div>
  <div class="headertitle">
<div class="title">genetree.h File Reference</div>  </div>
</div><!--header-->
<div class="contents">

<p>gene tree and species tree structures, for reconciliation etc. This is the high-level file with globally exposed functions/structures.  
<a href="#details">More...</a></p>
<div class="textblock"><code>#include &quot;<a class="el" href="upgma_8h_source.html">upgma.h</a>&quot;</code><br />
</div><div class="textblock"><div class="dynheader">
Include dependency graph for genetree.h:</div>
<div class="dyncontent">
<div class="center"><img src="genetree_8h__incl.png" border="0" usemap="#genetree_8h" alt=""/></div>
<map name="genetree_8h" id="genetree_8h">
<area shape="rect" id="node2" href="upgma_8h.html" title="UPGMA and bioNJ from (onedimensional representation of) distance matrices. " alt="" coords="593,80,665,107"/>
<area shape="rect" id="node3" href="topology__randomise_8h.html" title="Creation of random topologies and modification of existing ones through branch swapping. " alt="" coords="554,155,703,181"/>
<area shape="rect" id="node4" href="topology__distance_8h.html" title="branch length operations on topologies, including patristic distances " alt="" coords="615,229,754,256"/>
<area shape="rect" id="node29" href="prob__distribution_8h.html" title="Probability distribution functions and auxiliary mathematical functions from statistical package R..." alt="" coords="522,528,650,555"/>
<area shape="rect" id="node5" href="topology__common_8h.html" title="General&#45;purpose topology structures created from nexus_tree_struct (and low&#45;level functions) ..." alt="" coords="674,304,813,331"/>
<area shape="rect" id="node6" href="bipartition_8h.html" title="Unary/binary operators on arbitrarily&#45;sized bitstrings (strings of zeros and ones) like split biparti..." alt="" coords="759,901,848,928"/>
<area shape="rect" id="node24" href="distance__matrix_8h.html" title="distance matrix, that can be used in alignments and trees, and patristic&#45;distance based species dista..." alt="" coords="738,453,863,480"/>
<area shape="rect" id="node27" href="empirical__frequency_8h.html" title="Creates a histogram of a vector, ordered by frequency. " alt="" coords="613,379,762,405"/>
<area shape="rect" id="node7" href="lowlevel_8h.html" title="Lowest level header file. Header file for lowlevel.c. " alt="" coords="669,976,749,1003"/>
<area shape="rect" id="node25" href="hashtable_8h.html" title="double hashing open&#45;address hash table using strings as key – also has distance matrix, that can be used in alignments and trees " alt="" coords="756,752,845,779"/>
<area shape="rect" id="node26" href="hashfunctions_8h.html" title="Collections of hash functions for 32 and 64 bits, including one&#45;liners, murmurhash, and xxhash. " alt="" coords="746,827,861,853"/>
<area shape="rect" id="node28" href="char__vector_8h.html" title="list of strings (each string is a vector of chars) " alt="" coords="613,453,714,480"/>
<area shape="rect" id="node30" href="random__number_8h.html" title="Random number generation, with algorithms from the Gnu Scientific Library (GSL) and motivation from t..." alt="" coords="527,603,653,629"/>
<area shape="rect" id="node31" href="random__number__gen_8h_source.html" title="random_number_gen.h" alt="" coords="517,677,671,704"/>
</map>
</div>
</div><div class="textblock"><div class="dynheader">
This graph shows which files directly or indirectly include this file:</div>
<div class="dyncontent">
<div class="center"><img src="genetree_8h__dep__incl.png" border="0" usemap="#genetree_8hdep" alt=""/></div>
<map name="genetree_8hdep" id="genetree_8hdep">
<area shape="rect" id="node2" href="biomcmc_8h.html" title="biomcmc library interface to external programs, specific to super_sptree repo. " alt="" coords="5,80,92,107"/>
<area shape="rect" id="node3" href="reconciliation_8h.html" title="low&#45;level file for gene tree and species tree reconciliation. This file is hidden from user and conta..." alt="" coords="117,80,226,107"/>
<area shape="rect" id="node4" href="splitset__distances_8h.html" title="Low&#45;level functions that use only the split bipartitions of topologies – treating them as unrooted u..." alt="" coords="102,155,241,181"/>
</map>
</div>
</div>
<p><a href="genetree_8h_source.html">Go to the source code of this file.</a></p>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="nested-classes"></a>
Data Structures</h2></td></tr>
<tr class="memitem:"><td class="memItemLeft" align="right" valign="top">struct &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="structgenetree__struct.html">genetree_struct</a></td></tr>
<tr class="separator:"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:"><td class="memItemLeft" align="right" valign="top">struct &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="structspeciestree__struct.html">speciestree_struct</a></td></tr>
<tr class="separator:"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:"><td class="memItemLeft" align="right" valign="top">struct &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="structreconciliation__struct.html">reconciliation_struct</a></td></tr>
<tr class="memdesc:"><td class="mdescLeft">&#160;</td><td class="mdescRight">mapping between gene tree nodes (this) and (external) species tree nodes  <a href="structreconciliation__struct.html#details">More...</a><br /></td></tr>
<tr class="separator:"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:"><td class="memItemLeft" align="right" valign="top">struct &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="structsplitset__struct.html">splitset_struct</a></td></tr>
<tr class="separator:"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="typedef-members"></a>
Typedefs</h2></td></tr>
<tr class="memitem:ac8233c1da03b04b0b1b04633f5cb3122"><td class="memItemLeft" align="right" valign="top"><a id="ac8233c1da03b04b0b1b04633f5cb3122"></a>
typedef struct <a class="el" href="structgenetree__struct.html">genetree_struct</a> *&#160;</td><td class="memItemRight" valign="bottom"><b>genetree</b></td></tr>
<tr class="separator:ac8233c1da03b04b0b1b04633f5cb3122"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac5f4c18d18cb939cffa9c726b3563b73"><td class="memItemLeft" align="right" valign="top"><a id="ac5f4c18d18cb939cffa9c726b3563b73"></a>
typedef struct <a class="el" href="structspeciestree__struct.html">speciestree_struct</a> *&#160;</td><td class="memItemRight" valign="bottom"><b>speciestree</b></td></tr>
<tr class="separator:ac5f4c18d18cb939cffa9c726b3563b73"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6596cf66810a5c9aec2262bc9f996e4d"><td class="memItemLeft" align="right" valign="top"><a id="a6596cf66810a5c9aec2262bc9f996e4d"></a>
typedef struct <a class="el" href="structreconciliation__struct.html">reconciliation_struct</a> *&#160;</td><td class="memItemRight" valign="bottom"><b>reconciliation</b></td></tr>
<tr class="separator:a6596cf66810a5c9aec2262bc9f996e4d"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae74a526102c780cf0d428626c5ae443d"><td class="memItemLeft" align="right" valign="top"><a id="ae74a526102c780cf0d428626c5ae443d"></a>
typedef struct <a class="el" href="structsplitset__struct.html">splitset_struct</a> *&#160;</td><td class="memItemRight" valign="bottom"><b>splitset</b></td></tr>
<tr class="separator:ae74a526102c780cf0d428626c5ae443d"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="func-members"></a>
Functions</h2></td></tr>
<tr class="memitem:ac064bad64ced3e7de694ccfef830a91f"><td class="memItemLeft" align="right" valign="top"><a id="ac064bad64ced3e7de694ccfef830a91f"></a>
<a class="el" href="structgenetree__struct.html">genetree</a>&#160;</td><td class="memItemRight" valign="bottom"><b>new_genetree_speciestree_pair</b> (<a class="el" href="structtopology__struct.html">topology</a> gene, <a class="el" href="structtopology__struct.html">topology</a> species)</td></tr>
<tr class="separator:ac064bad64ced3e7de694ccfef830a91f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6fec27b0080a0d115a076ced4086bd49"><td class="memItemLeft" align="right" valign="top"><a id="a6fec27b0080a0d115a076ced4086bd49"></a>
<a class="el" href="structgenetree__struct.html">genetree</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="genetree_8h.html#a6fec27b0080a0d115a076ced4086bd49">new_genetree</a> (<a class="el" href="structtopology__struct.html">topology</a> gene, <a class="el" href="structspeciestree__struct.html">speciestree</a> sptre)</td></tr>
<tr class="memdesc:a6fec27b0080a0d115a076ced4086bd49"><td class="mdescLeft">&#160;</td><td class="mdescRight">Allocate space for new <a class="el" href="structgenetree__struct.html">genetree_struct</a>, given a gene topology and a specestree_struct. <br /></td></tr>
<tr class="separator:a6fec27b0080a0d115a076ced4086bd49"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6501ac5fa8b692fcc4c6c653e10ba851"><td class="memItemLeft" align="right" valign="top"><a id="a6501ac5fa8b692fcc4c6c653e10ba851"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><b>del_genetree</b> (<a class="el" href="structgenetree__struct.html">genetree</a> gtre)</td></tr>
<tr class="separator:a6501ac5fa8b692fcc4c6c653e10ba851"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a0296f1aa7c07a792dae428c4d7547bd1"><td class="memItemLeft" align="right" valign="top"><a id="a0296f1aa7c07a792dae428c4d7547bd1"></a>
<a class="el" href="structspeciestree__struct.html">speciestree</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="genetree_8h.html#a0296f1aa7c07a792dae428c4d7547bd1">new_speciestree</a> (<a class="el" href="structtopology__struct.html">topology</a> species, int *order_of_species_names)</td></tr>
<tr class="memdesc:a0296f1aa7c07a792dae428c4d7547bd1"><td class="mdescLeft">&#160;</td><td class="mdescRight">Allocate space for new <a class="el" href="structspeciestree__struct.html">speciestree_struct</a>, given a species topology and optionally the order of species names. <br /></td></tr>
<tr class="separator:a0296f1aa7c07a792dae428c4d7547bd1"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ac6ae213c357f8d36cfdae38f62ca49d7"><td class="memItemLeft" align="right" valign="top"><a id="ac6ae213c357f8d36cfdae38f62ca49d7"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><b>del_speciestree</b> (<a class="el" href="structspeciestree__struct.html">speciestree</a> sptre)</td></tr>
<tr class="separator:ac6ae213c357f8d36cfdae38f62ca49d7"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a490b42c86767967879c08d2383cffd9c"><td class="memItemLeft" align="right" valign="top"><a id="a490b42c86767967879c08d2383cffd9c"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="genetree_8h.html#a490b42c86767967879c08d2383cffd9c">genetree_speciestree_distances</a> (<a class="el" href="structgenetree__struct.html">genetree</a> gtre, <a class="el" href="structspeciestree__struct.html">speciestree</a> sptre)</td></tr>
<tr class="memdesc:a490b42c86767967879c08d2383cffd9c"><td class="mdescLeft">&#160;</td><td class="mdescRight">calculates all (discrete) distances and update min and max <br /></td></tr>
<tr class="separator:a490b42c86767967879c08d2383cffd9c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6dfc3e77d436f595fa650f2b7daae3b5"><td class="memItemLeft" align="right" valign="top"><a id="a6dfc3e77d436f595fa650f2b7daae3b5"></a>
int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="genetree_8h.html#a6dfc3e77d436f595fa650f2b7daae3b5">count_species_in_index_species_gene</a> (int *sp_id, int max_sp, int n_sp_id)</td></tr>
<tr class="memdesc:a6dfc3e77d436f595fa650f2b7daae3b5"><td class="mdescLeft">&#160;</td><td class="mdescRight">from gene-species map index, count number of distinct species represented <br /></td></tr>
<tr class="separator:a6dfc3e77d436f595fa650f2b7daae3b5"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a35e2a76279fd11bd044bfc7e352e3eda"><td class="memItemLeft" align="right" valign="top"><a id="a35e2a76279fd11bd044bfc7e352e3eda"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="genetree_8h.html#a35e2a76279fd11bd044bfc7e352e3eda">genetree_reconcile_speciestree</a> (<a class="el" href="structgenetree__struct.html">genetree</a> gtre, <a class="el" href="structspeciestree__struct.html">speciestree</a> sptre)</td></tr>
<tr class="memdesc:a35e2a76279fd11bd044bfc7e352e3eda"><td class="mdescLeft">&#160;</td><td class="mdescRight">&lt;debug function&gt;=""&gt; dups.loss, ils calculation; accepts unseen <a class="el" href="structspeciestree__struct.html">speciestree_struct</a> (i.e. updates mrca and pointers). Calls low-level hidden function. <br /></td></tr>
<tr class="separator:a35e2a76279fd11bd044bfc7e352e3eda"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aa3341a36618b5176b3be8122fe94f512"><td class="memItemLeft" align="right" valign="top"><a id="aa3341a36618b5176b3be8122fe94f512"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="genetree_8h.html#aa3341a36618b5176b3be8122fe94f512">genetree_dSPR_speciestree</a> (<a class="el" href="structgenetree__struct.html">genetree</a> gtre, <a class="el" href="structspeciestree__struct.html">speciestree</a> sptre, int level)</td></tr>
<tr class="memdesc:aa3341a36618b5176b3be8122fe94f512"><td class="mdescLeft">&#160;</td><td class="mdescRight">&lt;debug function&gt;=""&gt; dSPR (level &gt; 1), hdist (level &gt; 0), and RF distances; doesn't need to update sptree pointer <br /></td></tr>
<tr class="separator:aa3341a36618b5176b3be8122fe94f512"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><p>gene tree and species tree structures, for reconciliation etc. This is the high-level file with globally exposed functions/structures. </p>
</div></div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.13
</small></address>
</body>
</html>
