<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.13"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>biomcmc-lib: prob_distribution.h File Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectlogo"><img alt="Logo" src="_biomcmclib-thumbnail.png"/></td>
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">biomcmc-lib
   &#160;<span id="projectnumber">0.1</span>
   </div>
   <div id="projectbrief">low level library for phylogenetic analysis</div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.13 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Search');
</script>
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
$(function() {
  initMenu('',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
</script>
<div id="main-nav"></div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="dir_97aefd0d527b934f1d99a682da8fe6a9.html">lib</a></li>  </ul>
</div>
</div><!-- top -->
<div class="header">
  <div class="summary">
<a href="#nested-classes">Data Structures</a> &#124;
<a href="#typedef-members">Typedefs</a> &#124;
<a href="#func-members">Functions</a>  </div>
  <div class="headertitle">
<div class="title">prob_distribution.h File Reference</div>  </div>
</div><!--header-->
<div class="contents">

<p>Probability distribution functions and auxiliary mathematical functions from statistical package R.  
<a href="#details">More...</a></p>
<div class="textblock"><code>#include &quot;<a class="el" href="lowlevel_8h_source.html">lowlevel.h</a>&quot;</code><br />
<code>#include &quot;<a class="el" href="random__number_8h_source.html">random_number.h</a>&quot;</code><br />
</div><div class="textblock"><div class="dynheader">
Include dependency graph for prob_distribution.h:</div>
<div class="dyncontent">
<div class="center"><img src="prob__distribution_8h__incl.png" border="0" usemap="#prob__distribution_8h" alt=""/></div>
<map name="prob__distribution_8h" id="prob__distribution_8h">
<area shape="rect" id="node2" href="lowlevel_8h.html" title="Lowest level header file. Header file for lowlevel.c. " alt="" coords="669,453,749,480"/>
<area shape="rect" id="node19" href="random__number_8h.html" title="Random number generation, with algorithms from the Gnu Scientific Library (GSL) and motivation from t..." alt="" coords="712,80,839,107"/>
<area shape="rect" id="node20" href="random__number__gen_8h_source.html" title="random_number_gen.h" alt="" coords="705,155,859,181"/>
<area shape="rect" id="node21" href="hashtable_8h.html" title="double hashing open&#45;address hash table using strings as key – also has distance matrix, that can be used in alignments and trees " alt="" coords="732,229,821,256"/>
<area shape="rect" id="node22" href="hashfunctions_8h.html" title="Collections of hash functions for 32 and 64 bits, including one&#45;liners, murmurhash, and xxhash. " alt="" coords="714,304,829,331"/>
<area shape="rect" id="node23" href="bipartition_8h.html" title="Unary/binary operators on arbitrarily&#45;sized bitstrings (strings of zeros and ones) like split biparti..." alt="" coords="716,379,805,405"/>
</map>
</div>
</div><div class="textblock"><div class="dynheader">
This graph shows which files directly or indirectly include this file:</div>
<div class="dyncontent">
<div class="center"><img src="prob__distribution_8h__dep__incl.png" border="0" usemap="#prob__distribution_8hdep" alt=""/></div>
<map name="prob__distribution_8hdep" id="prob__distribution_8hdep">
<area shape="rect" id="node2" href="char__vector_8h.html" title="list of strings (each string is a vector of chars) " alt="" coords="300,80,401,107"/>
<area shape="rect" id="node14" href="topology__randomise_8h.html" title="Creation of random topologies and modification of existing ones through branch swapping. " alt="" coords="324,379,473,405"/>
<area shape="rect" id="node21" href="prob__distribution__aux_8h.html" title="Auxiliary (low level) functions for prob_distribution.c. " alt="" coords="594,80,750,107"/>
<area shape="rect" id="node3" href="nexus__common_8h.html" title="File handling functions for nexus format in general. " alt="" coords="57,229,183,256"/>
<area shape="rect" id="node10" href="empirical__frequency_8h.html" title="Creates a histogram of a vector, ordered by frequency. " alt="" coords="201,155,351,181"/>
<area shape="rect" id="node20" href="char__vector_8c.html" title="vector of strings (species names, leaf names, etc.) " alt="" coords="375,155,476,181"/>
<area shape="rect" id="node4" href="alignment_8h.html" title="File handling functions and calculation of distances for sequence data in nexus format. " alt="" coords="35,304,125,331"/>
<area shape="rect" id="node7" href="read__newick__trees_8h.html" title="Low&#45;level functions for reading newick strings. " alt="" coords="106,379,249,405"/>
<area shape="rect" id="node5" href="kmerhash_8h.html" title="k&#45;mer handling of DNA sequences, with hash transformation " alt="" coords="5,453,96,480"/>
<area shape="rect" id="node6" href="biomcmc_8h.html" title="biomcmc library interface to external programs, specific to super_sptree repo. " alt="" coords="171,603,258,629"/>
<area shape="rect" id="node8" href="newick__space_8h.html" title="Reads a list of trees in newick format and creates vector of topologies. " alt="" coords="121,453,237,480"/>
<area shape="rect" id="node9" href="topology__space_8h.html" title="Reads tree files in nexus format and creates a vector of topologies. " alt="" coords="91,528,215,555"/>
<area shape="rect" id="node11" href="topology__common_8h.html" title="General&#45;purpose topology structures created from nexus_tree_struct (and low&#45;level functions) ..." alt="" coords="207,229,345,256"/>
<area shape="rect" id="node19" href="empirical__frequency_8c.html" title="histogram of vectors, ordered by frequency. Also calculates MAP (modal) values. " alt="" coords="369,229,519,256"/>
<area shape="rect" id="node12" href="topology__distance_8h.html" title="branch length operations on topologies, including patristic distances " alt="" coords="213,304,352,331"/>
<area shape="rect" id="node13" href="parsimony_8h.html" title="binary and multistate parsimony matrices, together with bipartition extraction for MRP ..." alt="" coords="240,528,333,555"/>
<area shape="rect" id="node15" href="upgma_8h.html" title="UPGMA and bioNJ from (onedimensional representation of) distance matrices. " alt="" coords="363,453,435,480"/>
<area shape="rect" id="node16" href="genetree_8h.html" title="gene tree and species tree structures, for reconciliation etc. This is the high&#45;level file with globa..." alt="" coords="357,528,440,555"/>
<area shape="rect" id="node17" href="reconciliation_8h.html" title="low&#45;level file for gene tree and species tree reconciliation. This file is hidden from user and conta..." alt="" coords="344,603,453,629"/>
<area shape="rect" id="node18" href="splitset__distances_8h.html" title="Low&#45;level functions that use only the split bipartitions of topologies – treating them as unrooted u..." alt="" coords="329,677,468,704"/>
</map>
</div>
</div>
<p><a href="prob__distribution_8h_source.html">Go to the source code of this file.</a></p>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="nested-classes"></a>
Data Structures</h2></td></tr>
<tr class="memitem:"><td class="memItemLeft" align="right" valign="top">struct &#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="structdiscrete__sample__struct.html">discrete_sample_struct</a></td></tr>
<tr class="separator:"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="typedef-members"></a>
Typedefs</h2></td></tr>
<tr class="memitem:a97894f80e8538bd09fb2d0e3506fa358"><td class="memItemLeft" align="right" valign="top"><a id="a97894f80e8538bd09fb2d0e3506fa358"></a>
typedef struct <a class="el" href="structdiscrete__sample__struct.html">discrete_sample_struct</a> *&#160;</td><td class="memItemRight" valign="bottom"><b>discrete_sample</b></td></tr>
<tr class="separator:a97894f80e8538bd09fb2d0e3506fa358"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="func-members"></a>
Functions</h2></td></tr>
<tr class="memitem:a73fd29df2d61da07ed6bfc6f0fe2b531"><td class="memItemLeft" align="right" valign="top"><a id="a73fd29df2d61da07ed6bfc6f0fe2b531"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="prob__distribution_8h.html#a73fd29df2d61da07ed6bfc6f0fe2b531">biomcmc_discrete_gamma</a> (double alpha, double beta, double *rate, int nrates)</td></tr>
<tr class="memdesc:a73fd29df2d61da07ed6bfc6f0fe2b531"><td class="mdescLeft">&#160;</td><td class="mdescRight">Ziheng Yang's gamma discretization of rates. <br /></td></tr>
<tr class="separator:a73fd29df2d61da07ed6bfc6f0fe2b531"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a88c8490d706f874d7914a49e2ca57725"><td class="memItemLeft" align="right" valign="top">double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="prob__distribution_8h.html#a88c8490d706f874d7914a49e2ca57725">biomcmc_dexp_dt</a> (double d, double lambda, double m, <a class="el" href="lowlevel_8h.html#a97a80ca1602ebf2303258971a2c938e2">bool</a> log_p)</td></tr>
<tr class="memdesc:a88c8490d706f874d7914a49e2ca57725"><td class="mdescLeft">&#160;</td><td class="mdescRight">pdf of discrete truncated exponential (d is discrete, m is maximum value)  <a href="#a88c8490d706f874d7914a49e2ca57725">More...</a><br /></td></tr>
<tr class="separator:a88c8490d706f874d7914a49e2ca57725"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6d3934571fa5cc2765140ca91fed1f5b"><td class="memItemLeft" align="right" valign="top"><a id="a6d3934571fa5cc2765140ca91fed1f5b"></a>
double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="prob__distribution_8h.html#a6d3934571fa5cc2765140ca91fed1f5b">biomcmc_pexp_dt</a> (double d, double lambda, double m, <a class="el" href="lowlevel_8h.html#a97a80ca1602ebf2303258971a2c938e2">bool</a> log_p)</td></tr>
<tr class="memdesc:a6d3934571fa5cc2765140ca91fed1f5b"><td class="mdescLeft">&#160;</td><td class="mdescRight">cdf of discrete truncated exponential (d is discrete, m is maximum value): calculates P(D &lt;= d) <br /></td></tr>
<tr class="separator:a6d3934571fa5cc2765140ca91fed1f5b"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a9f9e09d86eb2e838a03a920ecc8ca9d1"><td class="memItemLeft" align="right" valign="top"><a id="a9f9e09d86eb2e838a03a920ecc8ca9d1"></a>
double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="prob__distribution_8h.html#a9f9e09d86eb2e838a03a920ecc8ca9d1">biomcmc_qexp_dt</a> (double p, double lambda, double m, <a class="el" href="lowlevel_8h.html#a97a80ca1602ebf2303258971a2c938e2">bool</a> log_p)</td></tr>
<tr class="memdesc:a9f9e09d86eb2e838a03a920ecc8ca9d1"><td class="mdescLeft">&#160;</td><td class="mdescRight">quantile of discrete truncated exponential, that is, finds d s.t. P(D &lt;= d) &gt;= p <br /></td></tr>
<tr class="separator:a9f9e09d86eb2e838a03a920ecc8ca9d1"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4e3e58f056e950b2dd28b851a03b480b"><td class="memItemLeft" align="right" valign="top"><a id="a4e3e58f056e950b2dd28b851a03b480b"></a>
double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="prob__distribution_8h.html#a4e3e58f056e950b2dd28b851a03b480b">biomcmc_dgamma</a> (double x, double alpha, double beta, <a class="el" href="lowlevel_8h.html#a97a80ca1602ebf2303258971a2c938e2">bool</a> log_p)</td></tr>
<tr class="memdesc:a4e3e58f056e950b2dd28b851a03b480b"><td class="mdescLeft">&#160;</td><td class="mdescRight">gamma density <br /></td></tr>
<tr class="separator:a4e3e58f056e950b2dd28b851a03b480b"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a03ca8bc4bf4d793a9491652b9b948fa9"><td class="memItemLeft" align="right" valign="top"><a id="a03ca8bc4bf4d793a9491652b9b948fa9"></a>
double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="prob__distribution_8h.html#a03ca8bc4bf4d793a9491652b9b948fa9">biomcmc_qgamma</a> (double p, double alpha, double beta, <a class="el" href="lowlevel_8h.html#a97a80ca1602ebf2303258971a2c938e2">bool</a> log_p)</td></tr>
<tr class="memdesc:a03ca8bc4bf4d793a9491652b9b948fa9"><td class="mdescLeft">&#160;</td><td class="mdescRight">gamma quantile (inverse CDF) <br /></td></tr>
<tr class="separator:a03ca8bc4bf4d793a9491652b9b948fa9"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:afbd4bb7864ab888bd1028f91e3a06a05"><td class="memItemLeft" align="right" valign="top"><a id="afbd4bb7864ab888bd1028f91e3a06a05"></a>
double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="prob__distribution_8h.html#afbd4bb7864ab888bd1028f91e3a06a05">biomcmc_pgamma</a> (double x, double alpha, double beta, <a class="el" href="lowlevel_8h.html#a97a80ca1602ebf2303258971a2c938e2">bool</a> log_p)</td></tr>
<tr class="memdesc:afbd4bb7864ab888bd1028f91e3a06a05"><td class="mdescLeft">&#160;</td><td class="mdescRight">computes the cummulative distribution function for the gamma distribution with shape parameter alpha and rate parameter beta, s.t. <img class="formulaInl" alt="$ E[X] = \alpha/\beta $" src="form_0.png"/>. The same as the (lower) incomplete gamma function. <br /></td></tr>
<tr class="separator:afbd4bb7864ab888bd1028f91e3a06a05"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a62f8429632008f0bbd6189cdb40ac348"><td class="memItemLeft" align="right" valign="top"><a id="a62f8429632008f0bbd6189cdb40ac348"></a>
double&#160;</td><td class="memItemRight" valign="bottom"><b>biomcmc_dnorm</b> (double x, double mu, double sigma, <a class="el" href="lowlevel_8h.html#a97a80ca1602ebf2303258971a2c938e2">bool</a> log_p)</td></tr>
<tr class="separator:a62f8429632008f0bbd6189cdb40ac348"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a8aaef10dee497c6190c6a879194df2ba"><td class="memItemLeft" align="right" valign="top"><a id="a8aaef10dee497c6190c6a879194df2ba"></a>
double&#160;</td><td class="memItemRight" valign="bottom"><b>biomcmc_qnorm</b> (double p, double mu, double sigma, <a class="el" href="lowlevel_8h.html#a97a80ca1602ebf2303258971a2c938e2">bool</a> log_p)</td></tr>
<tr class="separator:a8aaef10dee497c6190c6a879194df2ba"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a7f7ad72dce1e22c3955f36220c0c97ae"><td class="memItemLeft" align="right" valign="top"><a id="a7f7ad72dce1e22c3955f36220c0c97ae"></a>
double&#160;</td><td class="memItemRight" valign="bottom"><b>biomcmc_pnorm</b> (double x, double mu, double sigma, <a class="el" href="lowlevel_8h.html#a97a80ca1602ebf2303258971a2c938e2">bool</a> log_p)</td></tr>
<tr class="separator:a7f7ad72dce1e22c3955f36220c0c97ae"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a83ed55030da96560d9911ec8467561ca"><td class="memItemLeft" align="right" valign="top"><a id="a83ed55030da96560d9911ec8467561ca"></a>
double&#160;</td><td class="memItemRight" valign="bottom"><b>biomcmc_dlnorm</b> (double x, double meanlog, double sdlog, <a class="el" href="lowlevel_8h.html#a97a80ca1602ebf2303258971a2c938e2">bool</a> log_p)</td></tr>
<tr class="separator:a83ed55030da96560d9911ec8467561ca"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a87057af5495b4cf1b47825be7103bed3"><td class="memItemLeft" align="right" valign="top"><a id="a87057af5495b4cf1b47825be7103bed3"></a>
double&#160;</td><td class="memItemRight" valign="bottom"><b>biomcmc_qlnorm</b> (double p, double meanlog, double sdlog, <a class="el" href="lowlevel_8h.html#a97a80ca1602ebf2303258971a2c938e2">bool</a> log_p)</td></tr>
<tr class="separator:a87057af5495b4cf1b47825be7103bed3"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:abb5c9ac409793cdd4c1904fddaf30c3c"><td class="memItemLeft" align="right" valign="top"><a id="abb5c9ac409793cdd4c1904fddaf30c3c"></a>
double&#160;</td><td class="memItemRight" valign="bottom"><b>biomcmc_plnorm</b> (double x, double meanlog, double sdlog, <a class="el" href="lowlevel_8h.html#a97a80ca1602ebf2303258971a2c938e2">bool</a> log_p)</td></tr>
<tr class="separator:abb5c9ac409793cdd4c1904fddaf30c3c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae12b7e30b5f53414cd2442f019d7c2dd"><td class="memItemLeft" align="right" valign="top"><a id="ae12b7e30b5f53414cd2442f019d7c2dd"></a>
double&#160;</td><td class="memItemRight" valign="bottom"><b>biomcmc_dpois</b> (double x, double lambda, <a class="el" href="lowlevel_8h.html#a97a80ca1602ebf2303258971a2c938e2">bool</a> log_p)</td></tr>
<tr class="separator:ae12b7e30b5f53414cd2442f019d7c2dd"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a4dab97c31eb644182e8d517aeebdf272"><td class="memItemLeft" align="right" valign="top"><a id="a4dab97c31eb644182e8d517aeebdf272"></a>
double&#160;</td><td class="memItemRight" valign="bottom"><b>biomcmc_qpois</b> (double p, double lambda, <a class="el" href="lowlevel_8h.html#a97a80ca1602ebf2303258971a2c938e2">bool</a> log_p)</td></tr>
<tr class="separator:a4dab97c31eb644182e8d517aeebdf272"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a123b7ab4e4d9aa54bdc6b6d33db05e9e"><td class="memItemLeft" align="right" valign="top"><a id="a123b7ab4e4d9aa54bdc6b6d33db05e9e"></a>
double&#160;</td><td class="memItemRight" valign="bottom"><b>biomcmc_ppois</b> (double x, double lambda, <a class="el" href="lowlevel_8h.html#a97a80ca1602ebf2303258971a2c938e2">bool</a> log_p)</td></tr>
<tr class="separator:a123b7ab4e4d9aa54bdc6b6d33db05e9e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a0608aa98208da0150444253b4fb5d924"><td class="memItemLeft" align="right" valign="top"><a id="a0608aa98208da0150444253b4fb5d924"></a>
double&#160;</td><td class="memItemRight" valign="bottom"><b>biomcmc_rng_gamma</b> (double alpha, double beta)</td></tr>
<tr class="separator:a0608aa98208da0150444253b4fb5d924"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aadba5497e04e9f5457828b0c381c9cc9"><td class="memItemLeft" align="right" valign="top"><a id="aadba5497e04e9f5457828b0c381c9cc9"></a>
double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="prob__distribution_8h.html#aadba5497e04e9f5457828b0c381c9cc9">biomcmc_rng_norm</a> (double mu, double sigma)</td></tr>
<tr class="memdesc:aadba5497e04e9f5457828b0c381c9cc9"><td class="mdescLeft">&#160;</td><td class="mdescRight">Returns a random number from a Normal distribution N(mu, sigma^2) using 52 bits of precision. <br /></td></tr>
<tr class="separator:aadba5497e04e9f5457828b0c381c9cc9"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:acd2748dc39115fca65d43af9a872481a"><td class="memItemLeft" align="right" valign="top"><a id="acd2748dc39115fca65d43af9a872481a"></a>
double&#160;</td><td class="memItemRight" valign="bottom"><b>biomcmc_rng_lnorm</b> (double meanlog, double sdlog)</td></tr>
<tr class="separator:acd2748dc39115fca65d43af9a872481a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a676f8a3bd210cf4866352a6de3610533"><td class="memItemLeft" align="right" valign="top"><a id="a676f8a3bd210cf4866352a6de3610533"></a>
double&#160;</td><td class="memItemRight" valign="bottom"><b>biomcmc_rng_pois</b> (double mu)</td></tr>
<tr class="separator:a676f8a3bd210cf4866352a6de3610533"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a82eb3462c75be07d719e34eee13ded3e"><td class="memItemLeft" align="right" valign="top"><a id="a82eb3462c75be07d719e34eee13ded3e"></a>
double&#160;</td><td class="memItemRight" valign="bottom"><b>biomcmc_lgammafn</b> (double x, int *sgn)</td></tr>
<tr class="separator:a82eb3462c75be07d719e34eee13ded3e"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a3f7c83cbd1b3a3614a3bbc689459d431"><td class="memItemLeft" align="right" valign="top"><a id="a3f7c83cbd1b3a3614a3bbc689459d431"></a>
double&#160;</td><td class="memItemRight" valign="bottom"><b>biomcmc_gammafn</b> (double x)</td></tr>
<tr class="separator:a3f7c83cbd1b3a3614a3bbc689459d431"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a26c9720b4ab62ddfdd79d895eabe6228"><td class="memItemLeft" align="right" valign="top"><a id="a26c9720b4ab62ddfdd79d895eabe6228"></a>
double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="prob__distribution_8h.html#a26c9720b4ab62ddfdd79d895eabe6228">biomcmc_log1p</a> (double x)</td></tr>
<tr class="memdesc:a26c9720b4ab62ddfdd79d895eabe6228"><td class="mdescLeft">&#160;</td><td class="mdescRight">compute the relative error logarithm <img class="formulaInl" alt="$ \log(1 + x)$" src="form_1.png"/> (C99 standard) <br /></td></tr>
<tr class="separator:a26c9720b4ab62ddfdd79d895eabe6228"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:ae2e2c04e796a50f34dd4733fff2a0cfe"><td class="memItemLeft" align="right" valign="top"><a id="ae2e2c04e796a50f34dd4733fff2a0cfe"></a>
double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="prob__distribution_8h.html#ae2e2c04e796a50f34dd4733fff2a0cfe">biomcmc_log1pmx</a> (double x)</td></tr>
<tr class="memdesc:ae2e2c04e796a50f34dd4733fff2a0cfe"><td class="mdescLeft">&#160;</td><td class="mdescRight">accurate calculation of <img class="formulaInl" alt="$\log(1+x)-x$" src="form_2.png"/>, particularly for small x <br /></td></tr>
<tr class="separator:ae2e2c04e796a50f34dd4733fff2a0cfe"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a7a8f5ff734e5bd99c42feab68d987616"><td class="memItemLeft" align="right" valign="top"><a id="a7a8f5ff734e5bd99c42feab68d987616"></a>
double&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="prob__distribution_8h.html#a7a8f5ff734e5bd99c42feab68d987616">biomcmc_expm1</a> (double x)</td></tr>
<tr class="memdesc:a7a8f5ff734e5bd99c42feab68d987616"><td class="mdescLeft">&#160;</td><td class="mdescRight">compute <img class="formulaInl" alt="$ \exp(x) - 1$" src="form_3.png"/> accurately also when x is close to zero, i.e. <img class="formulaInl" alt="$|x| \ll 1 $" src="form_4.png"/> <br /></td></tr>
<tr class="separator:a7a8f5ff734e5bd99c42feab68d987616"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a1dc125c51e53403a651f7276f6c302e1"><td class="memItemLeft" align="right" valign="top"><a id="a1dc125c51e53403a651f7276f6c302e1"></a>
<a class="el" href="structdiscrete__sample__struct.html">discrete_sample</a>&#160;</td><td class="memItemRight" valign="bottom"><b>new_discrete_sample_from_frequencies</b> (double *prob, size_t size)</td></tr>
<tr class="separator:a1dc125c51e53403a651f7276f6c302e1"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a3565720f993b7e9072dc9b2188d03be9"><td class="memItemLeft" align="right" valign="top"><a id="a3565720f993b7e9072dc9b2188d03be9"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><b>del_discrete_sample</b> (<a class="el" href="structdiscrete__sample__struct.html">discrete_sample</a> g)</td></tr>
<tr class="separator:a3565720f993b7e9072dc9b2188d03be9"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a116f5d23fb7146de98ae4c909fca0c0f"><td class="memItemLeft" align="right" valign="top"><a id="a116f5d23fb7146de98ae4c909fca0c0f"></a>
size_t&#160;</td><td class="memItemRight" valign="bottom"><b>biomcmc_rng_discrete</b> (<a class="el" href="structdiscrete__sample__struct.html">discrete_sample</a> g)</td></tr>
<tr class="separator:a116f5d23fb7146de98ae4c909fca0c0f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a47a7d08ece437027b8ac3e7565204e1c"><td class="memItemLeft" align="right" valign="top"><a id="a47a7d08ece437027b8ac3e7565204e1c"></a>
double&#160;</td><td class="memItemRight" valign="bottom"><b>biomcmc_discrete_sample_pdf</b> (<a class="el" href="structdiscrete__sample__struct.html">discrete_sample</a> g, size_t k)</td></tr>
<tr class="separator:a47a7d08ece437027b8ac3e7565204e1c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a3a72ec6bb4c6a3f1e27eac7ffb819658"><td class="memItemLeft" align="right" valign="top"><a id="a3a72ec6bb4c6a3f1e27eac7ffb819658"></a>
double&#160;</td><td class="memItemRight" valign="bottom"><b>biomcmc_logspace_add</b> (double logx, double logy)</td></tr>
<tr class="separator:a3a72ec6bb4c6a3f1e27eac7ffb819658"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a76ecfdb1a63044d807a8866028deea5a"><td class="memItemLeft" align="right" valign="top"><a id="a76ecfdb1a63044d807a8866028deea5a"></a>
double&#160;</td><td class="memItemRight" valign="bottom"><b>biomcmc_logspace_sub</b> (double logx, double logy)</td></tr>
<tr class="separator:a76ecfdb1a63044d807a8866028deea5a"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:accfc39840db06eef9e7f9a70acef6636"><td class="memItemLeft" align="right" valign="top"><a id="accfc39840db06eef9e7f9a70acef6636"></a>
<a class="el" href="lowlevel_8h.html#a97a80ca1602ebf2303258971a2c938e2">bool</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="prob__distribution_8h.html#accfc39840db06eef9e7f9a70acef6636">biomcmc_isfinite</a> (double x)</td></tr>
<tr class="memdesc:accfc39840db06eef9e7f9a70acef6636"><td class="mdescLeft">&#160;</td><td class="mdescRight">check if number is between minus infinity and plus infinity, or NaN <br /></td></tr>
<tr class="separator:accfc39840db06eef9e7f9a70acef6636"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><p>Probability distribution functions and auxiliary mathematical functions from statistical package R. </p>
<p>Code derived from the <a href="http://www.r-project.org/">R project for Statistical Computing</a> version 2.9.1, available under the GPL license. It might be possible to use directly the standalone mathematical library "Rmath.h" from the R project instead of our implementation. The advantage would be a library updated more often than mine, at the cost of delegating to the guenomu user the installation and maintenance of the extra libraries (like GSL, for instance). In Debian this library can be installed through the package "r-mathlib". The original R library checks for several built-in compiler functions (like log1p(e) for calculating log(e+1) ) but I simply assume the compiler has none and reimplement them. The CDFs always assume the lower tail (upper tails must use 1. - lower tail) or equivalent.</p>
<p>The code for the discrete sampling comes from the GNU Scientific Library version 1.14 </p>
</div><h2 class="groupheader">Function Documentation</h2>
<a id="a88c8490d706f874d7914a49e2ca57725"></a>
<h2 class="memtitle"><span class="permalink"><a href="#a88c8490d706f874d7914a49e2ca57725">&#9670;&nbsp;</a></span>biomcmc_dexp_dt()</h2>

<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">double biomcmc_dexp_dt </td>
          <td>(</td>
          <td class="paramtype">double&#160;</td>
          <td class="paramname"><em>d</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">double&#160;</td>
          <td class="paramname"><em>lambda</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">double&#160;</td>
          <td class="paramname"><em>m</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype"><a class="el" href="lowlevel_8h.html#a97a80ca1602ebf2303258971a2c938e2">bool</a>&#160;</td>
          <td class="paramname"><em>log_p</em>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">

<p>pdf of discrete truncated exponential (d is discrete, m is maximum value) </p>
<p>pdf of discrete truncated exponential (d is discrete, m is maximum value) </p>

</div>
</div>
</div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.13
</small></address>
</body>
</html>
