<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.13"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>biomcmc-lib: splitset_distances.h File Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectlogo"><img alt="Logo" src="_biomcmclib-thumbnail.png"/></td>
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">biomcmc-lib
   &#160;<span id="projectnumber">0.1</span>
   </div>
   <div id="projectbrief">low level library for phylogenetic analysis</div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.13 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Search');
</script>
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
$(function() {
  initMenu('',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
</script>
<div id="main-nav"></div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><a class="el" href="dir_97aefd0d527b934f1d99a682da8fe6a9.html">lib</a></li>  </ul>
</div>
</div><!-- top -->
<div class="header">
  <div class="summary">
<a href="#func-members">Functions</a>  </div>
  <div class="headertitle">
<div class="title">splitset_distances.h File Reference</div>  </div>
</div><!--header-->
<div class="contents">

<p>Low-level functions that use only the split bipartitions of topologies &ndash; treating them as unrooted usually.  
<a href="#details">More...</a></p>
<div class="textblock"><code>#include &quot;<a class="el" href="reconciliation_8h_source.html">reconciliation.h</a>&quot;</code><br />
</div><div class="textblock"><div class="dynheader">
Include dependency graph for splitset_distances.h:</div>
<div class="dyncontent">
<div class="center"><img src="splitset__distances_8h__incl.png" border="0" usemap="#splitset__distances_8h" alt=""/></div>
<map name="splitset__distances_8h" id="splitset__distances_8h">
<area shape="rect" id="node2" href="reconciliation_8h.html" title="low&#45;level file for gene tree and species tree reconciliation. This file is hidden from user and conta..." alt="" coords="574,80,683,107"/>
<area shape="rect" id="node3" href="genetree_8h.html" title="gene tree and species tree structures, for reconciliation etc. This is the high&#45;level file with globa..." alt="" coords="587,155,670,181"/>
<area shape="rect" id="node4" href="upgma_8h.html" title="UPGMA and bioNJ from (onedimensional representation of) distance matrices. " alt="" coords="593,229,665,256"/>
<area shape="rect" id="node5" href="topology__randomise_8h.html" title="Creation of random topologies and modification of existing ones through branch swapping. " alt="" coords="554,304,703,331"/>
<area shape="rect" id="node6" href="topology__distance_8h.html" title="branch length operations on topologies, including patristic distances " alt="" coords="615,379,754,405"/>
<area shape="rect" id="node31" href="prob__distribution_8h.html" title="Probability distribution functions and auxiliary mathematical functions from statistical package R..." alt="" coords="522,677,650,704"/>
<area shape="rect" id="node7" href="topology__common_8h.html" title="General&#45;purpose topology structures created from nexus_tree_struct (and low&#45;level functions) ..." alt="" coords="674,453,813,480"/>
<area shape="rect" id="node8" href="bipartition_8h.html" title="Unary/binary operators on arbitrarily&#45;sized bitstrings (strings of zeros and ones) like split biparti..." alt="" coords="759,1051,848,1077"/>
<area shape="rect" id="node26" href="distance__matrix_8h.html" title="distance matrix, that can be used in alignments and trees, and patristic&#45;distance based species dista..." alt="" coords="738,603,863,629"/>
<area shape="rect" id="node29" href="empirical__frequency_8h.html" title="Creates a histogram of a vector, ordered by frequency. " alt="" coords="613,528,762,555"/>
<area shape="rect" id="node9" href="lowlevel_8h.html" title="Lowest level header file. Header file for lowlevel.c. " alt="" coords="669,1125,749,1152"/>
<area shape="rect" id="node27" href="hashtable_8h.html" title="double hashing open&#45;address hash table using strings as key â€“ also has distance matrix, that can be used in alignments and trees " alt="" coords="756,901,845,928"/>
<area shape="rect" id="node28" href="hashfunctions_8h.html" title="Collections of hash functions for 32 and 64 bits, including one&#45;liners, murmurhash, and xxhash. " alt="" coords="746,976,861,1003"/>
<area shape="rect" id="node30" href="char__vector_8h.html" title="list of strings (each string is a vector of chars) " alt="" coords="613,603,714,629"/>
<area shape="rect" id="node32" href="random__number_8h.html" title="Random number generation, with algorithms from the Gnu Scientific Library (GSL) and motivation from t..." alt="" coords="527,752,653,779"/>
<area shape="rect" id="node33" href="random__number__gen_8h_source.html" title="random_number_gen.h" alt="" coords="517,827,671,853"/>
</map>
</div>
</div>
<p><a href="splitset__distances_8h_source.html">Go to the source code of this file.</a></p>
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="func-members"></a>
Functions</h2></td></tr>
<tr class="memitem:a17b5845e332a9eda6a20373fb9d910b9"><td class="memItemLeft" align="right" valign="top"><a id="a17b5845e332a9eda6a20373fb9d910b9"></a>
<a class="el" href="structsplitset__struct.html">splitset</a>&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="splitset__distances_8h.html#a17b5845e332a9eda6a20373fb9d910b9">new_splitset_genespecies</a> (<a class="el" href="structtopology__struct.html">topology</a> gene, <a class="el" href="structtopology__struct.html">topology</a> species, <a class="el" href="structreconciliation__struct.html">reconciliation</a> rec)</td></tr>
<tr class="memdesc:a17b5845e332a9eda6a20373fb9d910b9"><td class="mdescLeft">&#160;</td><td class="mdescRight">Splitset structure for dSPR calculation (also allocates aux vectors) <br /></td></tr>
<tr class="separator:a17b5845e332a9eda6a20373fb9d910b9"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a0158ea2246ac03204cf66f0e6b95866c"><td class="memItemLeft" align="right" valign="top"><a id="a0158ea2246ac03204cf66f0e6b95866c"></a>
void&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="splitset__distances_8h.html#a0158ea2246ac03204cf66f0e6b95866c">del_splitset</a> (<a class="el" href="structsplitset__struct.html">splitset</a> split)</td></tr>
<tr class="memdesc:a0158ea2246ac03204cf66f0e6b95866c"><td class="mdescLeft">&#160;</td><td class="mdescRight">free memory allocated for splitset structure <br /></td></tr>
<tr class="separator:a0158ea2246ac03204cf66f0e6b95866c"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aec3b49df05b411245005015c601e6920"><td class="memItemLeft" align="right" valign="top"><a id="aec3b49df05b411245005015c601e6920"></a>
int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="splitset__distances_8h.html#aec3b49df05b411245005015c601e6920">dSPR_gene_species</a> (<a class="el" href="structtopology__struct.html">topology</a> gene, <a class="el" href="structtopology__struct.html">topology</a> species, <a class="el" href="structsplitset__struct.html">splitset</a> split)</td></tr>
<tr class="memdesc:aec3b49df05b411245005015c601e6920"><td class="mdescLeft">&#160;</td><td class="mdescRight">approximate dSPR between unrooted gene and species trees (leafset mapping) <br /></td></tr>
<tr class="separator:aec3b49df05b411245005015c601e6920"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a9fd60ac1e2b81b0e4bd8125a6ef2a3dd"><td class="memItemLeft" align="right" valign="top"><a id="a9fd60ac1e2b81b0e4bd8125a6ef2a3dd"></a>
int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="splitset__distances_8h.html#a9fd60ac1e2b81b0e4bd8125a6ef2a3dd">dSPR_gene_species_rf</a> (<a class="el" href="structtopology__struct.html">topology</a> gene, <a class="el" href="structtopology__struct.html">topology</a> species, <a class="el" href="structsplitset__struct.html">splitset</a> split)</td></tr>
<tr class="memdesc:a9fd60ac1e2b81b0e4bd8125a6ef2a3dd"><td class="mdescLeft">&#160;</td><td class="mdescRight">RF distance between unrooted gene and species trees (leafset mapping) <br /></td></tr>
<tr class="separator:a9fd60ac1e2b81b0e4bd8125a6ef2a3dd"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af7b75ecc862db6e6e486e6227c140017"><td class="memItemLeft" align="right" valign="top"><a id="af7b75ecc862db6e6e486e6227c140017"></a>
int&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="splitset__distances_8h.html#af7b75ecc862db6e6e486e6227c140017">dSPR_gene_species_hdist</a> (<a class="el" href="structtopology__struct.html">topology</a> gene, <a class="el" href="structtopology__struct.html">topology</a> species, <a class="el" href="structsplitset__struct.html">splitset</a> split)</td></tr>
<tr class="memdesc:af7b75ecc862db6e6e486e6227c140017"><td class="mdescLeft">&#160;</td><td class="mdescRight">h distance (edge disagreement assignment cost) between unrooted gene and species trees (leafset mapping) <br /></td></tr>
<tr class="separator:af7b75ecc862db6e6e486e6227c140017"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><p>Low-level functions that use only the split bipartitions of topologies &ndash; treating them as unrooted usually. </p>
<p>Use a "splitset" structure that copies the bipartition information of all nodes (so that original trees are untouched) and then modifies this splitset. These functions assume a gene tree (mul-tree) and a species tree (NOT mul-tree). Compared to guenomu and genefam-dist, I removed the simpler 'orthologous' functions since they assumed _same_leaves on both trees, which is not usual even without multrees. </p>
</div></div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.13
</small></address>
</body>
</html>
